### ISSUE TITLE
"Convert to interpolated string" for string + any object as warning

### ISSUE THREAD

**Issue by ufcpp** [dotnet/roslyn-analyzers#7108]
### Describe the problem you are trying to solve

Today, the following code was discussed on X (Twitter) in Japan.

```cs
Console.WriteLine(1 + "2"); // 12
```

This code, familiar to senior C# developers, seems strange to beginners. They say, "It doesn't look like a strongly typed language; it looks like JavaScript." Also, the following code looks confusing even for senior devs.

```cs
Console.WriteLine(1 + 1 + "1"); // 21
Console.WriteLine("1" + 1 + 1); // 111
```

Even today, code fixes are available for this code.

![image](https://github.com/dotnet/roslyn/assets/3760955/29142b7d-360c-4d87-86ae-b395cc1c7440)

### Describe suggestions on how to achieve the rule

What I would like to suggest here is to add an analyzer to this, so that the warning level can be changed in .editorconfig. I want to treat this as a warning.

**Comment by CyrusNajmabadi** [comment]
> What I would like to suggest here is to add an analyzer to this

You are welcome to add an analyzer for this :)

**Comment by CyrusNajmabadi** [comment]
@mavasani it doesn't seem like it fits any of the cases we include in Roslyn for our analyzers. 

I moved it here because it felt a little more like the old school "style cop" rules. 

If it's not a good fit for either project, we should just close this out.  A user could always write their own analyzer for this niche case.
